### ISSUE TITLE
Shift key not working properly in some situations

### ISSUE THREAD

**Issue by iCodeSometime** [microsoft/terminal#2771]
# Environment

```none
Windows build number: 10.0.18362.0
Windows Terminal version (if applicable): 0.4.2382.0

Any other software? wsl + emacs
```

# Steps to reproduce

1. Open emacs within wsl in Windows Terminal.
2. Type either of the following:
   - M-> (alt + greater than) to move to the end of the file
   - M-< (alt + less than) to move to the beginning of the file
3. Verify the key emacs is seeing, by typing C-h k (Ctrl+h, k) for describe key - then type either of the above key combinations.

# Expected behavior
M-> moves to the end of the file.
M-< moves to the beginning of the file.
C-h k M-> displays M-> as the command run.
C-h k M-< displays M-< as the command run.

# Actual behavior
M-> does not move to the end of the file.
M-< does not move to the beginning of the file.
C-h k M-> displays M-. as the command run.
C-h k M-< displays M-, as the command run.

**Comment by iCodeSometime** [comment]
To be clear, this all works properly when I use my wsl instance from outside of the new windows terminal app.

**Comment by zadjii-msft** [comment]
I'm moving this to the Terminal 1.0 milestone because it seems this only repros in the Windows Terminal. It _doesn't_ repro if you use WSL to launch cmd.exe, then launch `wsl emacs`. That set of steps would activate conpty. Since it doesn't repro with conhost acting as the terminal, we can rule our conpty as the source of the bug.

It'll still get fixed, but we don't need to worry about getting it into the OS for 20H1.

IIRC @lhecker did some work in this area previously.

**Comment by lhecker** [comment]
@zadjii-msft Yeah this problem is basically tangential to the AltGr issues.
We have the vkey and modifier states in the UWP code readily available, but when we map it for the old NT console subsystem we throw away the modifier states here: https://github.com/microsoft/terminal/blob/7128e873a449a6a4c2e5680de1eb82c395ed603d/src/cascadia/TerminalCore/Terminal.cpp#L225
This causes the logic here to malfunction where the assumption is made that the char data is already pretranslated: https://github.com/microsoft/terminal/blob/7128e873a449a6a4c2e5680de1eb82c395ed603d/src/terminal/input/terminalInput.cpp#L428
Due to this I added the circumvention logic for AltGr in the UWP side of things. I guess it's time to find a proper solution for this?

**Comment by lhecker** [comment]
I read up on this a bit and it appears to me that we can map the vkey to the pretranslated character using a combination of `MapVirtualKey(vkey, MAPVK_VK_TO_VSC)` with `ToUnicode(...)`.
Afterwards we should be able to remove the special handling for AltGr.

(I still don't quite understand the full architecture, but I keep wondering for what technical reason the `terminalInput` class is being used. If it's just for code sharing purposes, we should probably split the input handling up, so we don't have to rely on the above.)

**Comment by DHowett-MSFT** [comment]
Fixed in #2836.
